src/pathview/qgcodeprogramloader.h

```diff
@@ -62,16 +68,27 @@ class QGCodeProgramLoader : public QObject
     }
 
 signals:
+    void textChanged();
     void localFilePathChanged(QString arg);
     void localPathChanged(QString arg);
     void remotePathChanged(QString arg);
     void modelChanged(QGCodeProgramModel * arg);
     void loadingFinished();
     void loadingFailed();
+    void error(QString message);
 
 public slots:
+    void save(const QString &text);
+    void saveAs(const QString &localFilePath, const QString &text);
     void load();
 
+    void setText(const QString &arg) {
+        if (m_text != arg) {
```

Comment:
again, does not make sense to do this because of unnecessary comparison (100.000+ lines)